XenModel/Actions/VM/GpuAssignAction.cs

```diff
@@ -59,18 +57,31 @@ protected override void Run()
             foreach (VGPU vgpu in vm.Connection.ResolveAll(vm.VGPUs))
                 VGPU.destroy(Session, vgpu.opaque_ref);
 
-            if (gpu_group == null)  // The VM doesn't want a VGPU
+            if (vGpus == null || vGpus.Count == 0)  // The VM doesn't want a VGPU
                 return;
 
-            // Add the new VGPU
-            string device = "0";  // fixed at the moment, see PR-1060
-            Dictionary<string, string> other_config = new Dictionary<string,string>();
+            // Add the new VGPUs
+            int index = 0;
+            foreach (var vGpu in vGpus)
+            {
+                AddGpu(vm.Connection.Resolve(vGpu.GPU_group), vm.Connection.Resolve(vGpu.type), index + 11);

```

Comment:
Let's keep the temporary setting of device number here until XAPI part is ready.